Improved SDK Readme.md (#907)
Crossmint/crossmint-sdk: README.md

<div align="center">
  <h1 align="center">Crossmint SDK</h1>

  <p align="center">
    A comprehensive suite of tools for blockchain integration, authentication, and NFT functionality
    <br />
    <a href="https://docs.crossmint.com/sdk-reference/introduction"><strong>Explore the docs »</strong></a>
    <br />
    <br />

>     <a href="https://www.smarterwallet.dev/">View Demo</a>

Reviewer: I think I would link to playground.crossmint.com here

  </p>
</div>

<details>
  <summary>Table of Contents</summary>
  <ol>
    <li><a href="#about-the-project">About The Project</a></li>
    <li><a href="#package-overview">Package Overview</a></li>
    <li><a href="#getting-started">Getting Started</a></li>
    <li><a href="#building-sdk-locally">Building SDK Locally</a></li>
    <li><a href="#development">Development</a></li>
    <li><a href="#running-the-demo">Running the Demo</a></li>
    <li><a href="#making-package-changes">Making Package Changes</a></li>
    <li><a href="#publishing">Publishing</a></li>
    <li><a href="#migration-guides">Migration Guides</a></li>
    <li><a href="#documentation">Documentation</a></li>
    <li><a href="#contact">Contact</a></li>
  </ol>
</details>

## About The Project

![Crossmint SDK](https://github.com/user-attachments/assets/28d4ac5c-9373-45a3-969d-4dbf1d8200a8)

The Crossmint SDK is a powerful collection of packages designed to simplify blockchain integration for developers. Our goal is to make Web3 development accessible without requiring extensive blockchain experience or cryptocurrency knowledge.

### Why Crossmint?

> - 🚀 **Quick Integration**: Set up in just 5 minutes

Reviewer: ```suggestion
- 🚀 **Quick Integration**: all our tools have 5-min quickstarts
```

- 💻 **Developer-First**: Build end-to-end blockchain solutions without deep Web3 expertise
- 🔒 **Secure**: Enterprise-grade security for all blockchain interactions

> - 💳 **Fiat-First**: Enable users to participate in Web3 without cryptocurrency

Reviewer: drop this one

> - 🆓 **Free for Sellers**: No costs to integrate

Reviewer: ```suggestion
- 🆓 **Free to Start**: Start developing at no cost
```

### Core Features

> - 💳 Wallet-less credit card purchases for NFTs

Reviewer: ```suggestion
- 💳 NFT checkout: with credit card and cross-chain
```

- 🔐 Authentication and session management

> - 👛 Smart wallet integration

Reviewer: ```suggestion
- 👛 Embedded wallets
```

> - 📜 Verifiable credentials management
> - 🌐 Cross-platform compatibility

Reviewer: ```suggestion
- 📜 Verifiable credentials
```

## Package Overview

### Client-Side Packages

> - `@crossmint/client-sdk-react-ui`: React components and hooks for integrating blockchain, authentication, and NFT-related functionalities into your application.
> 
> - `@crossmint/client-sdk-base`: Core client-side functionality and essential building blocks for Crossmint integration.
> 
> - `@crossmint/client-sdk-auth`: Client-side authentication tools for managing user sessions and integrating various authentication methods.
> 
> - `@crossmint/client-sdk-verifiable-credentials`: Comprehensive tools for managing verifiable credentials, including verification, presentation, and decryption capabilities.
> 
> - `@crossmint/client-sdk-smart-wallet`: Smart wallet functionality featuring passkey support, auth providers (Privy, Dynamic), and blockchain providers (Viem account, EIP1193 compatible).
> 

Reviewer: ```suggestion
- `@crossmint/client-sdk-react-ui`: React SDK, with UI components.

- `@crossmint/client-sdk-auth`: Headless typescript SDK for managing auth and user profiles.

- `@crossmint/client-sdk-verifiable-credentials`: Headless typescript SDK for Verifiable Credentials.

- `@crossmint/client-sdk-smart-wallet`: Headless typescript SDK for smart wallets.

Other:

- `@crossmint/client-sdk-base`: Core client-side functionality and essential building blocks for Crossmint integration. Typically you don't need to integrate against this directly.

```

### Server-Side Packages

> - `@crossmint/server-sdk`: Server-side authentication and session management tools, specifically designed for SSR applications like Next.js.

Reviewer: ```suggestion
- `@crossmint/server-sdk`: Server-side SDK.
```

> ### Common Packages
> 
> - `@crossmint/common-sdk-base`: Shared utilities, enums, and common functionality across the SDK ecosystem.
> 
> - `@crossmint/common-sdk-auth`: Shared authentication utilities and tools used across different SDK packages.
> 

Reviewer: Drop this

```suggestion
```

### Demo Applications

> - `@crossmint/client-sdk-react-ui`: A NextJS application showcasing the full capabilities of the SDK, including authentication and smart wallet integration.

Reviewer: 1. I dont think this is the right package
2. I think there's more than one
3. We should add links to the folder

Author: we have a react demo but in my opinion it's not a great example of a demo app

Reviewer: ok!

## Getting Started

To get started with the SDK, install the packages you need into your project.

ie. 
`pnpm add @crossmint/client-sdk-smart-wallet`

Then, import the package you need.

ie.
`import { SmartWalletSDK } from "@crossmint/client-sdk-smart-wallet";`

For more information on how to use the SDK, see the [Crossmint SDK Documentation](https://docs.crossmint.com/sdk-reference/introduction).

## Building SDK locally

### Prerequisites

```shell
node >= 20
pnpm
```

### Installation

1. Clone the repository:
```shell
git clone https://github.com/Crossmint/crossmint-sdk.git
```

2. Install dependencies:
```shell
cd crossmint-sdk
pnpm install
```

3. Build all packages:
```shell
pnpm build
```

## Development

> Read the [Smart Wallets Demo (Next.js Starter Kit) README](./apps/wallets/smart-wallet/next/README.md) for more information on how to run the demo application.

Reviewer: Drop this Develpoment section

### Making Package Changes

When modifying packages locally:

1. Make your changes in the relevant package
2. Rebuild the modified package:
```shell
# Inside the package directory
pnpm build
```

3. Reinstall dependencies in your project:
```shell
# From the root directory
pnpm install
```

4. Restart the demo application to see your changes

## Publishing

1. Run `pnpm change:add` to select packages for update
2. Commit the generated changeset file
3. Merge your PR to `main`
4. A "Release packages" PR will be created automatically
5. Merge the release PR to publish to NPM

> ## Migration Guides
> 
> - [Migrating to 0.1.X](https://docs.google.com/document/d/14IKpjrij7kU7Dr0I7rZkf0PyDNbXiklx2v4GuzUrFbw/edit?usp=sharing)
> - [Migrating to 0.2.X](https://docs.google.com/document/d/1mA0W-iAs0nHHW0ANX0TfZ5qrzxPGxNchPj13W6cHc-Y/edit?usp=sharing)
> 

Reviewer: Drop 
```suggestion
```

- 💳 **Fiat-First**: Enable users to participate in Web3 without cryptocurrency
- 🆓 **Free to Start**: Start developing at no cost

### Core Features

- 💳 NFT checkout: with credit card and cross-chain
- 🔐 Authentication and session management
- 👛 Embedded wallets
- 📜 Verifiable credentials

## Package Overview

### Client-Side Packages

- `@crossmint/client-sdk-react-ui`: React SDK, with UI components.
	
- `@crossmint/client-sdk-auth`: Headless typescript SDK for managing auth and user profiles.
	
- `@crossmint/client-sdk-verifiable-credentials`: Headless typescript SDK for Verifiable Credentials.
	
- `@crossmint/client-sdk-smart-wallet`: Headless typescript SDK for smart wallets.
	
Other:
	
- `@crossmint/client-sdk-base`: Core client-side functionality and essential building blocks for Crossmint integration. Typically you don't need to integrate against this directly.

### Server-Side Packages

- `@crossmint/server-sdk`: Server-side SDK.

### Demo Applications

- `🔒 Smart Wallet + Auth Demo`: A NextJS application showcasing the full capabilities of the SDK, including authentication and smart wallet integration.

> Link to [Smart Wallets Demo (Next.js Starter Kit)](./apps/wallets/smart-wallet/next/README.md)

Reviewer: I meant the bullet point should have the link inside it, not to put it separately like here
